src/openfermion/ops/_binary_operator.py

```diff
@@ -0,0 +1,494 @@
+# symbolic binary class for decoder definitions (arxiv 1712.07067)
+
+import copy
+import numpy as np
+
+
+class SymbolicBinaryError(Exception):
+    pass
+
+
+class SymbolicBinary(object):
+    """
+        SymbolicBinaries are non-linear binary functions that are expressed analytically, they are
+        components of decoders that are used to rotate terms in qubit basis to fermionic basis. 
+        
+        A SymbolicBinary represents a sum of terms. Each term can consist of two types of binary variables:
+        (n,'W') where n stands for the qubit index, or (1,'1') for constant 1.
+         
+        The SymbolicBinary class overloads operations for manipulation of binary functions obeying binary rules.
+        1 + 1 = 0
+
+        Example:
+            .. code-block:: python
+
+                bin_fun = SymbolicBinary('1 + w1 w2 + w0 w1')
+                # Equivalently
+                bin_fun = SymbolicBinary([((1,'1'),)] + SymbolicBinary([((1,'W'),(2,'W')),((0,'W'),(1,'W'))])
+
+        """
+    actions = ('1', 'W')
+    action_strings = ('', 'W')
+    action_before_index = True
+    different_indices_commute = True
+
+    def __init__(self, term=None):
+        """
+        Initialize the SymbolicBinary based on term
+
+        Args:
```

Comment:
Since your function raises exceptions you should have "Raises:" in the doc strings in a similar way to how you have "Args:"